### PR TITLE
Checkpointing progress on 7536

### PR DESCRIPTION
**Pull request by jmthibault79**
AdminUserProfile is now at feature parity with the original AdminUser.  I'd like to merge this, with the understanding that there is work remaining.

TODO:
* remove AdminUser
* add the new Access Module Status table from the mocks
* add "dirty" indicators to changed-but-unsaved components
* add Enabled/Disabled toggle
* add API error messages from the previous version
* design and implement "has RT" and "has CT" indicators
* Open Question: replicate the "error" styling if the user has overspent the free credits?
* Open Question: where should we put the Other Role free-text box?

Jan 3 Checkpoint

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/routing/signed-in-app-routing.tsx [hunk 1]
```diff
@@ -108,6 +110,10 @@ export const SignedInRoutes = () => {
               guards={[authorityGuard(AuthorityGuardedAction.USER_ADMIN)]}>
       <UserAdminPage routeData={{title: 'User Admin', minimizeChrome: true}}/>
     </AppRoute>
+    <AppRoute exact path='/admin/users-tmp/:usernameWithoutGsuiteDomain'
```
Comment (jmthibault79) [inline ui/src/app/routing/signed-in-app-routing.tsx:113]: Temporary path to access the new page

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+}
+
+/**
+ * Present an ordered list of dollar options with the following values:
+ * $300 to $1000, in $100 increments
+ * $1000 to $10,000, in $500 increments
+ * Plus the user's current quota value, if it's not already one of these
+ */
+export const getFreeCreditLimitOptions = (freeTierDollarQuota?: number) => {
+  const START1 = 300;
```
Comment (NehaBroad) [inline ui/src/app/pages/admin/admin-user-common.tsx:87]: why start1 and not just start?

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    return null;
+  }
+}
+
+export const enableSave = (oldProfile: Profile, updatedProfile: Profile, errors): boolean =>
+  !errors && !fp.isEqual(oldProfile, updatedProfile);
+
+export const updateAccountProperties = async(oldProfile: Profile, updatedProfile: Profile): Promise<Profile> => {
+  const {username} = updatedProfile;
+  const request: AccountPropertyUpdate = {
```
Comment (NehaBroad) [inline ui/src/app/pages/admin/admin-user-common.tsx:125]: So if the attribute value is not different, we send null rather than existing unchanged value?

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    freeCreditsLimit: getUpdatedProfileValue(oldProfile, updatedProfile, 'freeTierDollarQuota'),
+    contactEmail: getUpdatedProfileValue(oldProfile, updatedProfile, 'contactEmail'),
+    affiliation: getUpdatedProfileValue(oldProfile, updatedProfile, 'verifiedInstitutionalAffiliation'),
+    accessBypassRequests: [],  // coming soon: RW-4958
+  };
+
+  return userAdminApi().updateAccountProperties(request);
+}
+
+export const DropdownWithLabel =
```
Comment (NehaBroad) [inline ui/src/app/pages/admin/admin-user-common.tsx:136]: Not part of this PR, but we should check if we can use this in other files/pages and move to a utils file

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+  return institutions
+    ? <DropdownWithLabel
+      dataTestId={'verifiedInstitution'}
+      label={'Verified institution'}
+      options={options}
+      initialValue={initialInstitutionShortName}
+      labelStyle={labelStyle}
+      dropdownStyle={dropdownStyle}
+      onChange={(event) => onChange(event)}/>
+    : null;
```
Comment (NehaBroad) [inline ui/src/app/pages/admin/admin-user-common.tsx:205]: should there be a spinner or error message if we cannot get institutions list

Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:205]: We have this on initial load.  Because UserAdminProfilePage is rendered with `withRoutingSpinner`, it has a spinner until we turn it off.  This happens on line 169, after we populate institutions.

It's true that errors in the institution response are not handled - do you think that's important to have here?

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 2 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:87]: my intention was START1+END1 and START2+END2.  I made a small update which I think improves this.

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:126]: right - this endpoint should only be called with changed values.  This allows us to only do the update work which is necessary, and also only record the _changes_ for the audit log.

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:137]: actually we may want to phase out Dropdown: https://precisionmedicineinitiative.atlassian.net/browse/RW-6023

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 5 above)
Comment (NehaBroad) [reply ui/src/app/pages/admin/admin-user-common.tsx:206]: So now that i am thinking about it,there are two scenarios:
1) The user has institution assigned: We get the value of the institution user belong to, has but not the complete list. : Thats ok we can at least show the user the assigned institution and thats it
2) User doesnt have institution assigned (maybe old users): Not sure how many of these exist and if they can access anything. But in such cases if we cannot show the institution list that will be an issue and we would need to show an error message

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 4 above)
Comment (NehaBroad) [reply ui/src/app/pages/admin/admin-user-common.tsx:137]: cool

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+
+/**
+ * Present an ordered list of dollar options with the following values:
+ * $300 to $1000, in $100 increments
+ * $1000 to $10,000, in $500 increments
+ * Plus the user's current quota value, if it's not already one of these
+ */
+export const getFreeCreditLimitOptions = (freeTierDollarQuota?: number) => {
+  const START1 = 300;
+  const END1 = 1000;
```
Comment (NehaBroad) [inline ui/src/app/pages/admin/admin-user-common.tsx:88]: nit: if end1 & end2 are not being used anywhere we could have its value 1001

**Review by NehaBroad — APPROVED** [review]

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 6 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:88]: I think this represents my intention better: we want to include the value 1000, so we specify "value+1"

File: ui/src/app/pages/admin/admin-user-common.tsx (hunk 5 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-common.tsx:206]: I'll add this to the TODOs in the description
